c/tskit/tables.c

```diff
@@ -7000,6 +7051,44 @@ tsk_table_collection_read_format_data(tsk_table_collection_t *self, kastore_t *s
     }
     memcpy(self->file_uuid, uuid, TSK_UUID_SIZE);
     self->file_uuid[TSK_UUID_SIZE] = '\0';
+
+    ret = kastore_containss(store, "metadata");
+    if (ret != 1 && ret != 0) {
+        ret = tsk_set_kas_error(ret);
+        goto out;
+    }
+    if (ret == 1) {
+        ret = kastore_gets_int8(
+            store, "metadata", (int8_t **) &metadata, (size_t *) &metadata_length);
```

Comment:
The cast here and on line 7079 are effectively what is happening in the table metadata already, but I think it isn't _strictly_ right. It is practically fine as it works in our tests, but it's my understanding that C99 only asks that `CHAR_BIT => 8` not `CHAR_BIT==8`

Reply:
Yeah, that is a potential issue. Probably we should just put in a ``assert(CHAR_BIT == 8)`` in the code somewhere central. It's not worth getting worried about more than that though.

Reply:
opened isssue to track, #645.